tests/tensorboard_test.py

```diff
@@ -0,0 +1,233 @@
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = onp.random.uniform(low=0., high=255., size=(30, 30, 3))
+    img = img.astype(onp.uint8)
+    summary_writer.image(tag='image_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, 'image_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    self.assertTrue(onp.allclose(img, expected_img.numpy()))
+
+  def test_summarywriter_image_float_pixel_values(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = onp.random.uniform(low=0., high=1., size=(30, 30, 3))
+    summary_writer.image(tag='image_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, 'image_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    # convert img to uint8 as the decoded image is of type uint8.
+    actual_img = tf.image.convert_image_dtype(image=img, dtype=onp.uint8)
+    self.assertTrue(onp.allclose(actual_img, expected_img.numpy()))
+
+  def test_summarywriter_2dimage_scaled(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = onp.random.uniform(low=0., high=255., size=(30, 30))
+    img = img.astype(onp.uint8)
+    summary_writer.image(tag='2dimage_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, '2dimage_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    # assert the image was increased in dimension
+    self.assertEqual(expected_img.shape, (30, 30, 3))
+
+  def test_summarywriter_single_channel_image_scaled(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    img = onp.random.uniform(low=0., high=255., size=(30, 30, 1))
+    img = img.astype(onp.uint8)
+    summary_writer.image(tag='2dimage_1channel_test', image=img, step=1)
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+
+    self.assertEqual(summary_value.tag, '2dimage_1channel_test')
+    expected_img = tf.image.decode_image(summary_value.tensor.string_val[2])
+    # assert the image was increased in dimension
+    self.assertEqual(expected_img.shape, (30, 30, 3))
+
+  def test_summarywriter_audio(self):
+    log_dir = tempfile.mkdtemp()
+    summary_writer = SummaryWriter(log_dir=log_dir)
+    audio = onp.random.uniform(low=-1., high=1., size=(2, 48000, 2))
+    summary_writer.audio(tag='audio_test', audiodata=audio, step=1)
+
+    summary_value = self.parse_and_return_summary_value(path=log_dir)
+    self.assertEqual(summary_value.tag, 'audio_test')
+
+    # Assert two audio files are parsed.
+    self.assertLen(summary_value.tensor.string_val, 4)
+
+    # Assert values.
+    expected_audio = \
```

Comment:
Are the names `audio` and `expected_audio` swapped maybe?

`expected_audio` should be the one we generate and write (so line 139) and the audio should be the one we read from summaries (so line 150). Or I am really misreading this?

I think this also applies to some other tests.

Reply:
yup, that's true. My bad, made the changes in rest of the occurrences as well. Thanks for catching this!